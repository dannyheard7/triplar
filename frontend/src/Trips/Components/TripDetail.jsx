import React from "react";
import Moment from 'moment';


export default class TripDetail extends React.Component {
    render () {
        const trip = this.props.trip;
        Moment.locale('en');

        return (
            <div id="trip-detail">
                <h2 id="trip-name">{trip.name}</h2>
                <p>{Moment(trip.startDate).format('Do MMMM') } - {Moment(trip.endDate).format('Do MMMM') }</p>
                <p className="created-by">Created by: {trip.created_by}</p>
                <p>Locations: {trip.locations.map(({city}, index) =>
                    (index ? ', ' : '') + city.name + (city.country ? ", " + city.country.name : ""))}</p>
            </div>
        )
    }
}

TripDetail.defaultProps = {
    trip: null
};
