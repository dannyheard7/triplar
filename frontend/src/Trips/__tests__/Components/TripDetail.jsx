import React from "react";
import {shallow} from "enzyme";
import ShallowRenderer from 'react-test-renderer/shallow';
const renderer = new ShallowRenderer();

import TripDetail from "Trips/Components/TripDetail";

const faker = require('faker');

describe('<TripDetail />', () => {
    describe('with no trip prop', () => {
        test.skip('displays no <ListGroupItem /> component', () => {
        });
    });

    describe('with trip prop', () => {
        let trip = null;

        beforeAll(function () {
            trip = {'id': faker.random.number(), 'name': faker.random.word(), 'created_by': faker.internet.email(),
            'locations': [faker.address.city()]};
        });

        test('renders correctly', () => {
            trip = {'id': 1, 'name': "test", 'created_by': "email@example.com", 'locations': ["Bristol"],
                    'start_date': "2018-08-10", "end_date": "2018-08-12"};
            const result = renderer.render(<TripDetail trip={trip}/>);
            expect(result).toMatchSnapshot();
        });

        test('displays the trip name', () => {
            const container = shallow(<TripDetail trip={trip}/>);
            expect(container.find("#trip-name").text()).toEqual(trip['name']);
        });

       test('displays the trip creator', () => {
            const container = shallow(<TripDetail trip={trip}/>);
            expect(container.find("#created-by").text()).toEqual('Created by: ' + trip['created_by']);
        });

        test('updates when trip prop changes', () => {
            const container = shallow(<TripDetail trip={trip}/>);
            let newTrip = {
                'id': faker.random.number(),
                'name': faker.random.word(),
                'created_by': faker.internet.email(),
                'locations': [faker.address.city()]
            };
            container.setProps({trip: newTrip});
            expect(container.find("#trip-name").text()).toEqual(newTrip['name']);
        })
    });
});