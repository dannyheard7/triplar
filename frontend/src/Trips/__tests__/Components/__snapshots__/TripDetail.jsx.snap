// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripDetail /> with trip prop renders correctly 1`] = `
<div
  id="trip-detail"
>
  <h2
    id="trip-name"
  >
    test
  </h2>
  <p>
    10th August
     - 
    12th August
  </p>
  <p
    className="created-by"
  >
    Created by: 
    email@example.com
  </p>
  <p>
    Locations: 
    undefined, undefined
  </p>
</div>
`;
